src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/CornerRadius/CornerRadiusEffect.shared.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.ComponentModel;
+using System.Linq;
+using Xamarin.Forms;
+using Xamarin.Forms.Shapes;
+
+namespace Xamarin.CommunityToolkit.Effects
+{
+	public class CornerRadiusEffect : NullEffect
+	{
+		public static readonly BindableProperty CornerRadiusProperty = BindableProperty.CreateAttached(
+			nameof(CornerRadius),
+			typeof(CornerRadius),
+			typeof(CornerRadiusEffect),
+			default(CornerRadius),
+			propertyChanged: TryGenerateEffect);
+
+		public static CornerRadius GetCornerRadius(BindableObject? bindable)
+			=> (CornerRadius)(bindable?.GetValue(CornerRadiusProperty) ?? throw new ArgumentNullException(nameof(bindable)));
+
+		public static void SetCornerRadius(BindableObject? bindable, CornerRadius value)
+			=> bindable?.SetValue(CornerRadiusProperty, value);
+
+		static void TryGenerateEffect(BindableObject? bindable, object oldValue, object newValue)
+		{
+			if (bindable is not VisualElement elementView)
+				return;
+
+			var cornerRadiusEffects = elementView.Effects.OfType<CornerRadiusEffect>();
+
+			if (GetCornerRadius(bindable) == default)
+			{
+				foreach (var cornerRadiusEffect in cornerRadiusEffects.ToArray())
+					elementView.Effects.Remove(cornerRadiusEffect);
+
+				return;
+			}
+
+			if (!cornerRadiusEffects.Any())
+				elementView.Effects.Add(new CornerRadiusEffect());
```

Comment:
Let's create an array and reduce one LINQ call.
```suggestion
			var cornerRadiusEffects = elementView.Effects.OfType<CornerRadiusEffect>().ToArray();

			if (GetCornerRadius(bindable) == default)
			{
				foreach (var cornerRadiusEffect in cornerRadiusEffects)
					elementView.Effects.Remove(cornerRadiusEffect);

				return;
			}

			if (cornerRadiusEffects.Length == 0)
				elementView.Effects.Add(new CornerRadiusEffect());
```

Reply:
Maybe the same fix here? https://github.com/xamarin/XamarinCommunityToolkit/blob/c6e30062681b5733870ac87932753cb9fb7dd10f/src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Shadow/ShadowEffect.shared.cs#L106